app/controllers/admin/resources_controller.rb

```diff
@@ -23,12 +24,19 @@ def create
     end
 
     def update
-      if @resource.update_attributes(resource_params)
-        redirect_to admin_conference_resources_path(conference_id: @conference.short_title),
-                    notice: 'Resource successfully updated.'
-      else
-        flash.now[:error] = "Resource update failed: #{@resource.errors.full_messages.join('. ')}."
-        render :edit
+      successful_update = check_successful_update params[:increment_used_resource_flag]
+      respond_to do |format|
+        if successful_update
+          format.html do
+            redirect_to admin_conference_resources_path(conference_id: @conference.short_title),
+                        notice: 'Resource successfully updated.'
+          end
+          flash.now[:notice] = get_successful_update_flash params[:increment_used_resource_flag]
```

Comment:
>Why do we need a function to define our flash message with a simple if statement?

I thought modularizing update action into smaller functions would make it more readable. I'm trying my best to make it readable by implementing different things. Will remove it.